### PR TITLE
[Runtime] Extend ObjC bridging casts to convert NSError to Error when nested in a container type.

### PR DESCRIPTION
**Pull request by mikeash**
rdar://problem/39349762

### CODE REVIEW

**Comment by mikeash** [comment]
@millenomi Sure! Do we have any tests of error bridging for that currently? If not, is it just a matter of taking the bits of `test/stdlib/ErrorBridged.swift` that will run without ObjC and putting them in a file that doesn't require ObjC interop?

**Comment by mikeash** [comment]
Answering my own question, SCF has its own `NSError` tests but doesn't cover this case. I added a test in this PR: https://github.com/apple/swift-corelibs-foundation/pull/1619

It passes so I think we're good!

**Comment by mikeash** [comment]
@swift-ci please test

File: stdlib/public/runtime/ErrorObject.mm [hunk 1]
```diff
@@ -521,19 +501,49 @@ NSInteger getErrorCode(const OpaqueValue *error,
...
   return false;
 }
 
+bool
+swift::tryDynamicCastNSErrorToValue(OpaqueValue *dest,
+                                    OpaqueValue *src,
+                                    const Metadata *srcType,
+                                    const Metadata *destType,
+                                    DynamicCastFlags flags) {
+  auto CFErrorTypeID = SWIFT_LAZY_CONSTANT(CFErrorGetTypeID());
```
Comment (jckarter) [inline stdlib/public/runtime/ErrorObject.mm]: This appears to be unused.

**Comment by mikeash** [comment]
@swift-ci please test

**Pull request merged**